controllers/post.contollers.js

```diff
@@ -4,13 +4,22 @@ module.exports = {
   findAll: async (req, res) => {
     db.post
       .find(req.query)
+      .populate("user_id")
       .then((dbPost) => res.json(dbPost))
       .catch((err) => res.status(422).json(err));
   },
   findById: async (req, res) => {
     db.post
       .findById(req.params.id)
-      .populate("answers")
+      // .populate("answers")
+      .populate({
+        path: "answers",
+        model: "Answer",
+        populate: {
+          path: "user",
+          model: "User",
```

Comment:
I think this works for now, but as your user model grows and has more fields you might not need to populate all fields on every answer.  Only the ones you really need (maybe image for avatar, username, profile link, etc) and not the entire object

https://mongoosejs.com/docs/populate.html#field-selection